### PR TITLE
Bugfix/5391

### PR DESCRIPTION
**Pull request by rupalid**
Implementation of GetAddressListMembers API Request and Response class
```

 <urn1:GetAddressListMembersRequest limit="500" offset="5"   id="some_id" countOnly="false"/>

<GetAddressListMembersResponse xmlns="urn:zimbraAccount"> <alm total="100">
<member>abc</member>
<member>def</member>
</alm>
</GetAddressListMembersResponse>

```

Pagination is not implemented and will be done in phase 2.
Verified test cases in test rail.
Verified that GAL and LDAP queries are correctly executed

### CODE REVIEW

File: common/src/java/com/zimbra/common/soap/AccountConstants.java [hunk 1]
```diff
@@ -401,6 +409,7 @@
     public static final String E_BOSH_URL = "boshURL";
     public static final String E_SUBSCRIPTION = "sub";
     public static final String E_IS_TRACKING_IMAP = "isTrackingIMAP";
+    public static final String E_ALM = "alm";
```
Comment (desouzas) [inline common/src/java/com/zimbra/common/soap/AccountConstants.java]: Dupe of the expanded `E_ADDRESS_LIST_MEMBER` constant above

File: soap/src/java/com/zimbra/soap/account/message/GetAddressListMemberResponse.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+     */
+    @XmlAttribute(name = AccountConstants.A_MORE /* more */, required = false)
+    private ZmBoolean more;
+
+    /**
+     * @zm-api-field-tag total
+     * @zm-api-field-description total number of distribution lists (not
+     *                           affected by limit/offset)
+     */
+    @XmlAttribute(name = AccountConstants.A_TOTAL /* total */, required = false)
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/account/message/GetAddressListMemberResponse.java]: The total seems to always be returned in the implementation, is it not required?

File: store/docs/soap.txt [hunk 3]
```diff
@@ -4826,3 +4826,18 @@ Returns list of address lists.
             ...
          </AddressLists>
       </GetAllAddressListsResponse>
+
+-----------------------------
+GetAdderssListMembers
```
Comment (desouzas) [inline store/docs/soap.txt]: typo

File: store/src/java/com/zimbra/cs/gal/GalSearchControl.java [hunk 4]
```diff
@@ -501,7 +502,40 @@ private boolean doLocalGalAccountSearch(Account galAcct) {
...
+                    if (hit instanceof ContactHit) {
+                        ContactHit chit = (ContactHit)hit;
+                        l.add(chit.getContact().getEmailAddresses().get(0));
+                       
+                    }
+                    num++;
+                    if (num == mParams.getLimit())
+                        break;
+                }
+//                callback.setSortBy(zqr.getSortBy().toString());
```
Comment (desouzas) [inline store/src/java/com/zimbra/cs/gal/GalSearchControl.java]: Remove commented code.

File: soap/src/java/com/zimbra/soap/account/message/GetAddressListMemberResponse.java (hunk 2 above)
Comment (rupalid) [reply soap/src/java/com/zimbra/soap/account/message/GetAddressListMemberResponse.java]: Fixed

File: store/docs/soap.txt (hunk 3 above)
Comment (rupalid) [reply store/docs/soap.txt]: Fixed

File: store/docs/soap.txt [hunk 5]
```diff
@@ -4826,3 +4826,18 @@ Returns list of address lists.
...
          </AddressLists>
       </GetAllAddressListsResponse>
+
+-----------------------------
+GetAddressListMembers
+
+API to get the members of an address list
+
+Request:
+<urn1:GetAddressListMembersRequest limit="{max result items}" offset="{offset from which result should be returned}"   id="{address list id}" countOnly="0|1{if only total account is returned}"/>
```
Comment (chinmay15) [inline store/docs/soap.txt]: remove urn1

**Pull request closed without merging**